forge/forge/op/loss.py

```diff
@@ -9,6 +10,50 @@
 from .reduce import ReduceSum, ReduceAvg
 
 
+def validate_shapes(min_dim=None, max_dim=None):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, predictions, labels, *args, **kwargs):
+            assert (
+                predictions.ndim() == labels.ndim()
+            ), f"Number of dimensions must match. Got {predictions.ndim()} and {labels.ndim()}."
+            if min_dim is not None:
+                assert (
+                    predictions.ndim() >= min_dim
+                ), f"Number of dimensions must be at least {min_dim}. Got {predictions.ndim()}."
+            if max_dim is not None:
+                assert (
+                    predictions.ndim() <= max_dim
+                ), f"Number of dimensions must be at most {max_dim}. Got {predictions.ndim()}."
```

Comment:
nit: for clarity, add tensor identifiers to the assert prints.
Like "Nr of dimensions of predictions and labels must match", "Nr of dims of predictions tensor must be..."